### ISSUE TITLE
Problems with layout alignment for crossed facet with independent scales

### ISSUE THREAD

**Issue by kanitw** [vega/vega#949]
From this [spec])(https://pastebin.com/AJafGPcp), even though I have set `layout.align` to `"all"` or `"each"`, it seems like the alignment either not consider empty cell or not consider non-empty cell's axes.

![vega_editor](https://user-images.githubusercontent.com/111269/28808539-a60a5a96-7631-11e7-8ba4-b7fdea25cbf2.png)

Even worse, if I don't set "align", it seems totally broken. 

![vega_editor](https://user-images.githubusercontent.com/111269/28808547-b7c06d02-7631-11e7-9e55-cfbcb2d64985.png)

To give an idea what I tried to make with the Vega spec above, here is the original VL spec.

```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "data": {"values": [
    {"r": "r1", "c": "c1", "a": "a1", "b": "b1"},
    {"r": "r1", "c": "c1", "a": "a2", "b": "b2"},
    {"r": "r2", "c": "c2", "a": "a1", "b": "b1"},
    {"r": "r3", "c": "c2", "a": "a3", "b": "b2"}
  ]},
  "facet": {
    "row": {"field": "r", "type": "nominal"},
    "column": {"field": "c", "type": "nominal"}
  },
  "spec": {
    "mark": "rect",
    "encoding": {
      "y": {"field": "b", "type": "nominal"},
      "x": {"field": "a", "type": "nominal"}
    }
  },
  "resolve": {
    "x": {
      "scale": "independent"
    },
    "y": {
      "scale": "independent"
    }
  }
}
```

**Comment by jheer** [comment]
The layout `bounds` parameter is set to "flush" by default, which performs layout based on group x/y/width/height alone. Did you try setting it to "full" instead, which incorporates the full bounds with axes, legends, etc?

**Comment by kanitw** [comment]
In the provided spec, we already use `bounds: "full"`

**Comment by jheer** [comment]
Thanks! This is indeed a bug that only became apparent with these empty cell conditions. Fixed in vega/vega-view@e10c55c6fae26550a80e3f1040e0916029eeab74, will be included in subsequent releases.
